nextflow/MaveDB/nf_modules/licences.nf

```diff
@@ -0,0 +1,74 @@
+process get_licence {
+  // Check if files are open access (error out if not)
+
+  errorStrategy 'ignore'
+
+  input:  path mappings
+  output: tuple path(mappings), path('LICENCE.txt')
+
+  """
+  #!/usr/bin/env python3
+  import urllib.request
+  import json
+
+  urn = "urn:mavedb:${mappings.simpleName}"
+  url = f"https://api.mavedb.org/api/v1/scoresets/{urn}"
+  res = urllib.request.urlopen(url).read()
+  res = json.loads(res)
+
+  f = open("LICENCE.txt", "w")
+  f.write(res['license']['shortName'])
+  f.close()
+  """
+}
+
+workflow get_files_by_licence {
+  take:
+    files
+    licences
+  main:
+    get_licence( files )
+    // Warn about discarded files
+    get_licence.out.
+      filter{ !licences.contains(it.last().text) }.
+      subscribe{
+        println "NOTE: Discarded ${it.first().simpleName} based on licence ${it.last().text}"
+      }
+    // Return files with matching licences
+    filtered = get_licence.out.
+      filter{ licences.contains(it.last().text) }.
+      map{ it.first() }
+  emit:
+    filtered
+}
+
+def split_by_mapping_type (files) {
```

Comment:
this function does not seems to be related to license - should we have a separate module for it?  or, if we change the module to name to `fetch` (or something similar - see below comment) move the `get_license` to different module (e.g. - `utils`)?